### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by angelotadeucci**
- **InteractActorPacket**
    - Only write extra int if `InteractActorType` is **Extractor** or **Gathering**;
- **FieldObjectPacket**
    - Small fix on **LoadPlayer** packet;
    - Added packet to remove player from map;
- **FieldManager**
    - Added `FieldObjectPacket.RemovePlayer()`;
- **InventoryController**
   - `Consume()` don't remove item if amount changed is greater than amount owned;
   - Removed `Update()` function;
   - Updated `MasteryHandler` and `RequestItemUseMultipleHandler` to use `Consume()`.
- **Dotnet-format**

### CODE REVIEW

**Review by AlanMorel — APPROVED** [review]

File: MapleServer2/PacketHandlers/Game/RequestItemUseMultipleHandler.cs [hunk 1]
```diff
@@ -39,43 +40,27 @@ public override void Handle(GameSession session, PacketReader packet)
...
-
+                Item item = kvp.Value;
                 // Do nothing if box has no data stored
                 if (item.Content.Count <= 0)
                 {
                     break;
                 }
 
-                for (int i = opened; i < amount; i++)
+                for (int i = 0; i < amount; i++)
```
Comment (rhydev) [inline MapleServer2/PacketHandlers/Game/RequestItemUseMultipleHandler.cs:54]: This is incorrect, without keeping track of number of opened boxes it'll keep resetting the amount of boxes to open when it goes to the next stack of boxes.

File: MapleServer2/Packets/FieldObjectPacket.cs [hunk 2]
```diff
@@ -21,19 +22,19 @@ public static Packet LoadPlayer(IFieldObject<Player> fieldPlayer)
...
             pWriter.WriteByte();
             pWriter.Write(fieldPlayer.Coord);
             pWriter.WriteShort(player.Levels.Level);
-            pWriter.WriteEnum(player.Job); // short
-            pWriter.WriteEnum(player.JobCode);
-            pWriter.WriteInt();
-            pWriter.WriteInt();
-            pWriter.WriteInt();
+            pWriter.WriteShort((short) player.Job);
+            pWriter.WriteShort((short) player.JobCode);
```
Comment (rhydev) [inline MapleServer2/Packets/FieldObjectPacket.cs:34]: This should be using `WriteEnum`, not sure why changed.

**Review by rhydev — CHANGES_REQUESTED** [review]

**Review by Sparkymod — COMMENTED** [review]
should be good, as soon the requested changes are resolve

File: MapleServer2/Packets/FieldObjectPacket.cs (hunk 2 above)
Comment (angelotadeucci) [reply MapleServer2/Packets/FieldObjectPacket.cs:34]: Because it's needed to write a short and most other places Job is used as int.

**Review by rhydev — APPROVED** [review]

**Pull request merged**